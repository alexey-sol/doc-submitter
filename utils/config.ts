export default {
    appName: process.env.NEXT_PUBLIC_APP_NAME ?? "Doc Submitter",
    nodeEnv: process.env.NODE_ENV ?? "production",
    webUrl: process.env.NEXT_PUBLIC_WEB_URL ?? "0.0.0.0:3000",
};
