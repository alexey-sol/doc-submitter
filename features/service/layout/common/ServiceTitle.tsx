import { ButtonBase, Typography } from "@mui/material";
import { SxProps } from "@mui/system";
import React from "react";
import config from "@/utils/config";

interface Props {
    handleClick?: () => void;
    rootSx?: SxProps;
}

const ServiceLogo = ({ handleClick, rootSx }: Props) => (
    <ButtonBase
        disabled={!handleClick}
        disableRipple
        onClick={handleClick}
        sx={rootSx}
    >
        <Typography variant="h5">
            {config.appName}
        </Typography>
    </ButtonBase>
);

ServiceLogo.defaultProps = {
    handleClick: undefined,
    rootSx: undefined,
} as Partial<Props>;

export default ServiceLogo;
