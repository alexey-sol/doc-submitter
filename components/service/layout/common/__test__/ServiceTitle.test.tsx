import React from "react";
import ReactDOM from "react-dom";
import ReactTestRenderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import ServiceTitle from "../ServiceTitle";
import config from "@/utils/config";

afterEach(cleanup);

it("should render ServiceTitle successfully", () => {
    const container = document.createElement("div");
    ReactDOM.render(<ServiceTitle />, container);
});

it("should be clickable if handleClick is provided", () => {
    const handleClickMock = jest.fn();

    const result = render(<ServiceTitle handleClick={handleClickMock} />);
    const elem = result.getByRole("button");

    fireEvent.click(elem);

    expect(handleClickMock).toBeCalledTimes(1);
    expect(elem).not.toBeDisabled();
});

it("should not be clickable if handleClick is not provided", () => {
    const result = render(<ServiceTitle handleClick={undefined} />);
    const elem = result.getByRole("button");

    fireEvent.click(elem);

    expect(elem).toBeDisabled();
});

it("should contain the app name", () => {
    const result = render(<ServiceTitle handleClick={undefined} />);
    const elem = result.getByRole("button");

    expect(elem).toHaveTextContent(config.appName);
});

it("should match snapshot", () => {
    const renderer = ReactTestRenderer.create(<ServiceTitle />).toJSON();
    expect(renderer).toMatchSnapshot();
});
