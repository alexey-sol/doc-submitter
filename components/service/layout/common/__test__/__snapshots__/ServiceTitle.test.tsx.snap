// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<button
  className="MuiButtonBase-root Mui-disabled css-10d1a0h-MuiButtonBase-root"
  disabled={true}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  <h5
    className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
  >
    Doc Submitter
  </h5>
</button>
`;
